tchannel/tornado/connection.py

```diff
@@ -701,3 +633,162 @@ def _request_timed_out(self, req_id, req_ttl, future):
         # while.
         future.set_exception(errors.TimeoutError())
         self._request_tombstones.add(req_id, req_ttl)
+
+
+class Reader(object):
+
+    def __init__(self, io_stream, capacity=64):
+        self.queue = tornado.queues.Queue(capacity)
+        self.filling = False
+        self.io_stream = io_stream
+
+    def fill(self):
+        self.filling = True
+
+        io_loop = IOLoop.current()
+
+        def keep_reading(f):
+            if f.exception():
+                return log(f.exception())
```

Comment:
I'm pretty sure this is preserving existing behavior, probably because the connection is in a bad state if this happens. I'll set `filling = False` which will have the effect of re-starting the reader as needed.